site/public/js/drag-and-drop.js

```diff
@@ -848,6 +848,75 @@ function gatherInputAnswersByType(type){
     return input_answers;
 }
 
+/**
+ * @param versions_used
+ * @param versions_allowed
+ * @param csrf_token
+ * @param gradeable_id
+ * @param user_id
+ * @param regrade
+ * @param regrade_all
+ * @param submissions
+ * @param regrade_all_students
+ * @param regrade_all_students_all
+ * differences between regrade, regrade_all, regrade_all_students and regrade_all_students_all
+ * regrade - regrade the active version for one selected student who submitted a certain gradeable
+ * regrade_all - regrade every version for one selected student who submitted a certain gradeable
+ * regrade_all_students - regrade the active version for every student who submitted a certain gradeable
+ * regrade_all_students_all regrade every version for every student who submitted a certain gradeable
+ */
+function handleRegrade(versions_used, csrf_token, gradeable_id, user_id, regrade = false, regrade_all=false, regrade_all_students = false, regrade_all_students_all = false) {
+    var submit_url = buildCourseUrl(['gradeable', gradeable_id, 'regrade']);
+    //don't redirect on success
+    var return_url = ''
+    var formData = new FormData();
+    formData.append('csrf_token', csrf_token);
+    formData.append('user_id', user_id);
+    formData.append('regrade', regrade);
+    formData.append('regrade_all', regrade_all);
+    formData.append('version_to_regrade', versions_used);
+    formData.append('regrade_all_students', regrade_all_students);
+    formData.append('regrade_all_students_all', regrade_all_students_all);
+    $.ajax({
+        url: submit_url,
+        data: formData,
+        processData: false,
+        headers : {
+            Accept: "application/json"
+        },
+        contentType: false,
+        type: 'POST',
+        success: function(data) {
+            try {
+                data = JSON.parse(data);
+                if (data['status'] === 'success') {
+                    window.location.href = return_url;
```

Comment:
Wouldn't a window.location.reload() make more sense? It seems you are simply refreshing the page